cirq-ft/cirq_ft/infra/gate_with_registers.py

```diff
@@ -51,6 +51,60 @@ def __repr__(self):
         return f'cirq_ft.Register(name="{self.name}", shape={self.shape})'
 
 
+def total_bits(registers: Iterable[Register]) -> int:
+    return sum(reg.total_bits() for reg in registers)
+
+
+def split_qubits(
+    registers: Iterable[Register], qubits: Sequence[cirq.Qid]
+) -> Dict[str, NDArray[cirq.Qid]]:  # type: ignore[type-var]
+    qubit_regs = {}
+    base = 0
+    for reg in registers:
+        qubit_regs[reg.name] = np.array(qubits[base : base + reg.total_bits()]).reshape(reg.shape)
+        base += reg.total_bits()
+    return qubit_regs
+
+
+def merge_qubits(
+    registers: Iterable[Register],
+    **qubit_regs: Union[cirq.Qid, Sequence[cirq.Qid], NDArray[cirq.Qid]],
+) -> List[cirq.Qid]:
+    ret: List[cirq.Qid] = []
+    for reg in registers:
+        assert (
+            reg.name in qubit_regs
+        ), f"All qubit registers must be present. {reg.name} not in qubit_regs"
```

Comment:
I know this is just moved code, but since I'm looking at it: this should probably be an `if` and a `ValueError` rather than a beefy assert.

Reply:
changed to `ValueError` and added new tests